jerry-core/parser/js/lexer.cpp

```diff
@@ -206,9 +207,11 @@ lexer_create_token_for_charset (jsp_token_type_t tt, /**< token type */
     return create_token_from_lit (tt, lit);
   }
   lit = lit_create_literal_from_utf8_string (converted_str_p, new_length);
-  JERRY_ASSERT (lit->get_type () == LIT_STR_T
-                || lit->get_type () == LIT_MAGIC_STR_T
-                || lit->get_type () == LIT_MAGIC_STR_EX_T);
+  record_type_t type = rcs_record_get_type (lit);
```

Comment:
If the type is used only for the assert, isn't there a warning on unused variable in release build?